partitura/score.py

```diff
@@ -2773,20 +2810,228 @@ class RomanNumeral(TimedObject):
         See parameters
     """
 
-    def __init__(self, text):
+    def __init__(self, text, inversion=None, local_key=None, primary_degree=None, secondary_degree=None, quality=None):
+        super().__init__(text)
+        self.text = text
+        self.accepted_qualities = ('7', 'aug', 'aug6', 'aug7', 'dim', 'dim7', 'hdim7', 'maj', 'maj7', 'min', 'min7')
+        self.has_seven = "7" in text
+        self.inversion = inversion if inversion is not None else self._process_inversion()
+        self.local_key = local_key if local_key is not None else self._process_local_key()
+        self.primary_degree = primary_degree if primary_degree is not None else self._process_primary_degree()
+        self.secondary_degree = secondary_degree if secondary_degree is not None else self._process_secondary_degree()
+        self.quality = quality if quality is not None and quality in self.accepted_qualities else self._process_quality()
+        # only process the root note if the roman numeral is valid
+        if self.local_key and self.primary_degree and self.secondary_degree and self.quality and self.inversion:
+            self.root = self.find_root_note()
+            self.bass_note = self.find_bass_note()
+
+    def _process_inversion(self):
+        """Find the inversion of the roman numeral from the text"""
+        # The inversion should be right after the roman numeral.
+        # If there is no inversion, return 0
+        numeric_indications_in_text = re.findall(r'\d+', self.text)
+        if len(numeric_indications_in_text) > 0:
+            inversion_state = int(numeric_indications_in_text[0])
+            if inversion_state == 2:
+                self.has_seven = True
+                return 3
+            elif inversion_state == 43:
+                self.has_seven = True
+                return 2
+            elif inversion_state == 64:
+                self.has_seven = False
+                return 2
+            elif inversion_state == 6:
+                self.has_seven = False
+                return 1
+            elif inversion_state == 65:
+                self.has_seven = True
+                return 1
```

Comment:
add an else statement if the inversion is not part of the previous options

Reply:
this long if-elif can be made with a dictionary to be easier to read and maintain.
The key of the dictionary is the text, and the value is a tuple (has_seven,inversion).